core/executor/src/wasmi_execution.rs

```diff
@@ -357,8 +357,9 @@ fn call_in_wasm_module(
 		},
 		|res, memory| {
 			if let Some(I64(r)) = res {
+				let r = u64::from_le(r as u64);
```

Comment:
As discussed below, I don't think that this makes sense.

Reply:
If you run this code on a big endian machine, it makes sense, because otherwise it would be incorrect.